mlflow/models/evaluation/default_evaluator.py

```diff
@@ -261,28 +270,42 @@ def _do_plot(**kwargs):
 }
 
 
-def _evaluate_custom_metric(index, custom_metric, eval_df, builtin_metrics):
+_CustomMetric = namedtuple("_CustomMetric", ["function", "name", "index"])
```

Comment:
Have you considered making a callable class with name and index attributes? It may look cleaner

Reply:
Oh that's true. I went with `namedtuple` because Corey suggested it to me, and also I saw that there's another similar usage of `namedtuple` in this file for [`_Curve`](https://github.com/mlflow/mlflow/blob/561eeffc8dd9a62585821065d5ee8c0fbc186e24/mlflow/models/evaluation/default_evaluator.py#L166).

Reply:
I opted instead to use `typing.NamedTuple`. It is a much cleaner approach imo.